pandas/_libs/tslibs/timedeltas.pyi

```diff
@@ -67,10 +67,19 @@ class Timedelta(timedelta):
     def __abs__(self) -> timedelta: ...
     def __mul__(self, other: float) -> timedelta: ...
     def __rmul__(self, other: float) -> timedelta: ...
-    @overload
+    # error: Signature of "__floordiv__" incompatible with supertype "timedelta"
+    @overload  # type: ignore[override]
     def __floordiv__(self, other: timedelta) -> int: ...
     @overload
-    def __floordiv__(self, other: int) -> timedelta: ...
+    def __floordiv__(self, other: int | float) -> timedelta: ...
+    @overload
+    def __floordiv__(self, other: np.ndarray) -> npt.NDArray[np.timedelta64]: ...
+    @overload
+    def __rfloordiv__(self, other: timedelta) -> int: ...
+    @overload
+    def __rfloordiv__(self, other: int | float) -> timedelta: ...
```

Comment:
wouldnt this case raise?

Reply:
Yes! I removed this overload and added one for `None | NaTType -> NaTType` and string also seems to be allowed.